### PR TITLE
Exclude AFKed players from the sleep threshold

### PR DESCRIPTION
**Pull request by harbar20**

### CODE REVIEW

**Comment by lukec11** [comment]
@ifvictr thoughts on this based on the previous converstation? 

> when i first added this, i kinda wanted to discourage afk servers, where everyone just hops on to afk instead of actually playing

**Comment by nice6599** [comment]
looks good to me.
i think this is beyond the scope of the pr but this may introduce a situation in which one person sleeps and the other people on the server (who are not really afk) type `/afk` and the server sleeps.

**Comment by harbar20** [comment]
I agree with @ifvictr to an extent, because while we don't want to make the server an AFK server, it also severely inconveniences active players when they're trying to skip the night.

**Comment by nice6599** [comment]
> severely inconveniences active players when they're trying to skip the night.

it definitely does inconvenience people

File: src/main/java/com/hackclub/hccore/listeners/SleepListener.java [hunk 1]
```diff
@@ -158,7 +159,15 @@ private int getSleepingPlayers(World world) {
...
+        int afkSum = 0;
+        for (Player player : world.getPlayers()) {
+            PlayerData playerData = this.plugin.getDataManager().getData(player);
+            if (playerData.isAfk()) {
+                afkSum++;
+            }
+        }
         return (int) Math.ceil(world.getPlayers().size()
-                * this.plugin.getConfig().getDouble("settings.skip-sleep-threshold"));
+                * this.plugin.getConfig().getDouble("settings.skip-sleep-threshold")) - afkSum;
```
Comment (nice6599) [inline src/main/java/com/hackclub/hccore/listeners/SleepListener.java:171]: you need to subtract `afkSum` before multiplying by `settings.skip-sleep-threshold`

File: src/main/java/com/hackclub/hccore/listeners/SleepListener.java [hunk 2]
```diff
@@ -158,7 +159,15 @@ private int getSleepingPlayers(World world) {
     }
 
     private int getMinSleepingPlayersNeeded(World world) {
+        //Get the number of AFK players
```
Comment (ifvictr) [inline src/main/java/com/hackclub/hccore/listeners/SleepListener.java:162]: Add a space after the //

File: src/main/java/com/hackclub/hccore/listeners/SleepListener.java [hunk 3]
```diff
@@ -158,7 +159,15 @@ private int getSleepingPlayers(World world) {
...
+        int afkSum = 0;
```
Comment (ifvictr) [inline src/main/java/com/hackclub/hccore/listeners/SleepListener.java:163]: Rename to `afkPlayersCount`

**Comment by harbar20** [comment]
Another thing to Nick's point: people already leave the game to speed up the night; why not make it easier to do that with the /afk?

**Pull request merged**